ambry-store/src/test/java/com.github.ambry.store/StoreManagerTest.java

```diff
@@ -0,0 +1,201 @@
+ */
+
+package com.github.ambry.store;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.clustermap.MockDataNodeId;
+import com.github.ambry.clustermap.MockPartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+import java.util.Set;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+
+public class StoreManagerTest {
+  private static final Random RANDOM = new Random();
+  private MockClusterMap clusterMap;
+
+  /**
+   * Startup the {@link MockClusterMap} for a test.
+   * @throws IOException
+   */
+  @Before
+  public void initializeCluster()
+      throws IOException {
+    clusterMap = new MockClusterMap(false, 1, 3, 3);
+  }
+
+  /**
+   * Cleanup the {@link MockClusterMap} after a test.
+   * @throws IOException
+   */
+  @After
+  public void cleanupCluster()
+      throws IOException {
+    clusterMap.cleanup();
+  }
+
+  /**
+   * Test that stores on a disk without a valid mount path are not started.
+   * @throws Exception
+   */
+  @Test
+  public void mountPathNotFoundTest()
+      throws Exception {
+    MockDataNodeId dataNode = clusterMap.getDataNodes().get(0);
+    List<ReplicaId> replicas = clusterMap.getReplicaIds(dataNode);
+    List<String> mountPaths = dataNode.getMountPaths();
+    String mountPathToDelete = mountPaths.get(RANDOM.nextInt(mountPaths.size()));
+    deleteDirectory(new File(mountPathToDelete));
+    StoreManager storeManager = createAndStartStoreManager(replicas);
+    for (ReplicaId replica : replicas) {
+      if (replica.getMountPath().equals(mountPathToDelete)) {
+        try {
+          storeManager.getStore(replica.getPartitionId());
+          fail("Should not have been able to get store for partition with deleted mount path");
+        } catch (StoreException e) {
+          assertEquals("Unexpected error code", StoreErrorCodes.Store_Not_Started, e.getErrorCode());
+        }
+      } else {
+        storeManager.getStore(replica.getPartitionId());
+      }
+    }
+    shutdownAndAssertStoresInaccessible(storeManager, replicas);
+  }
+
+  /**
+   * Tests that {@link StoreManager} can start even when certain stores cannot be started. Checks that these stores
+   * are not accessible. We can make the replica path non-readable to induce a store starting failure.
+   * @throws Exception
+   */
+  @Test
+  public void storeStartFailureTest()
```

Comment:
can you add a test for this scenario: all stores in a particular disk failed to start. Just for completeness sake